source/common/config/grpc_managed_mux_subscription_impl.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "envoy/api/v2/core/base.pb.h"
+#include "envoy/config/subscription.h"
+#include "envoy/event/dispatcher.h"
+#include "envoy/grpc/async_client.h"
+
+#include "common/config/grpc_mux_impl.h"
+#include "common/config/grpc_mux_subscription_impl.h"
+#include "common/config/utility.h"
+
+namespace Envoy {
+namespace Config {
+
+template <class ResourceType>
+class GrpcManagedMuxSubscriptionImpl : public Config::Subscription<ResourceType> {
```

Comment:
We should make https://github.com/envoyproxy/envoy/blob/master/source/common/config/grpc_subscription_impl.h a subclass of this as discussed on Slack.